includes/class-wc-legacy-rest-api-plugin.php

```diff
@@ -58,14 +68,57 @@ public static function on_plugin_activated() {
         }
     }
 
+    /**
+     * Add the "legacy REST API and HPOS are incompatible" admin notice if needed.
+     * 
+     * @returns bool True if the notice has been added, false otherwise.
+     */
+    private static function maybe_add_hpos_incompatibility_admin_notice() {
+        if( 'yes' !== get_option( 'woocommerce_custom_orders_table_enabled' ) || self::user_has_dismissed_admin_notice( 'legacy_rest_api_is_incompatible_with_hpos' ) ) {
```

Comment:
Should this use the `OrderUtils` check instead of accessing option directly?

Reply:
Makes sense. I have pushed the change.